distreqx/distributions/mvn_tri.py

```diff
@@ -0,0 +1,108 @@
+"""MultivariateNormalTri distribution."""
+
+from typing import Optional
+
+import jax.numpy as jnp
+from jaxtyping import Array
+
+from ..bijectors import DiagLinear, TriangularLinear
+from .mvn_from_bijector import MultivariateNormalFromBijector
+
+
+def _check_parameters(loc: Optional[Array], scale_tri: Optional[Array]) -> None:
+    """Checks that the inputs are correct."""
+    if loc is None and scale_tri is None:
+        raise ValueError("At least one of `loc` and `scale_tri` must be specified.")
+
+    if loc is not None and loc.ndim < 1:
+        raise ValueError("The parameter `loc` must have at least one dimension.")
+
+    if scale_tri is not None and scale_tri.ndim < 2:
+        raise ValueError(
+            f"The parameter `scale_tri` must have at least two dimensions, but "
+            f"`scale_tri.shape = {scale_tri.shape}`."
+        )
+
+    if scale_tri is not None and scale_tri.shape[-1] != scale_tri.shape[-2]:
+        raise ValueError(
+            f"The parameter `scale_tri` must be a (batched) square matrix, but "
+            f"`scale_tri.shape = {scale_tri.shape}`."
+        )
+
+    if loc is not None:
+        num_dims = loc.shape[-1]
+        if scale_tri is not None and scale_tri.shape[-1] != num_dims:
+            raise ValueError(
+                f"Shapes are not compatible: `loc.shape = {loc.shape}` and "
+                f"`scale_tri.shape = {scale_tri.shape}`."
+            )
+
+
+class MultivariateNormalTri(MultivariateNormalFromBijector):
+    """Multivariate normal distribution on `R^k`.
+
+    The `MultivariateNormalTri` distribution is parameterized by a `k`-length
+    location (mean) vector `b` and a (lower or upper) triangular scale matrix `S`
+    of size `k x k`. The covariance matrix is `C = S @ S.T`.
+    """
+
+    _scale_tri: Array
+    _is_lower: bool
+
+    def __init__(
+        self,
+        loc: Optional[Array] = None,
+        scale_tri: Optional[Array] = None,
+        is_lower: bool = True,
+    ):
+        """Initializes a MultivariateNormalTri distribution.
+
+        **Arguments:**
+
+        - `loc`: Mean vector of the distribution of shape `k` (can also be a batch of
```

Comment:
I would also remove explicit references of batch from the doc strings. It will probably implicitly work, but it's always possible to batch by vmaping the constructor.